document/core/exec/runtime.rst

```diff
@@ -678,51 +701,70 @@ This definition allows to index active labels surrounding a :ref:`branch <syntax
 the following syntax of *throw contexts* is defined, as well as associated structural rules:
 
 .. math::
    \begin{array}{llll}
    \production{(throw contexts)} & \XT &::=&
      \val^\ast~[\_]~\instr^\ast \\ &&|&
      \LABEL_n\{\instr^\ast\}~\XT~\END \\ &&|&
-     \CATCHN_n\{\instr^\ast\}~\XT~\END \\ &&|&
+     \CAUGHTadm\{\tagaddr~\val^\ast\}~\XT~\END \\ &&|&
      \FRAME_n\{F\}~\XT~\END \\
    \end{array}
 
-Throw contexts allow matching the program context around a throw instruction up to the nearest enclosing :math:`\CATCHN` handler, thereby selecting the exception handler responsible for an exception.
+Throw contexts allow matching the program context around a throw instruction up to the innermost enclosing |CATCHadm| or |DELEGATEadm|, thereby selecting the exception |handler| responsible for an exception, if one exists.
 
 .. note::
-   For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try block <exec-try>` would be as follows.
+   Contrary to block contexts, throw contexts don't skip over handlers.
+
+   Since handlers are not included above, there is always a unique maximal throw context to match the reduction rules.
+
+   |CAUGHTadm| blocks do not represent active handlers. Instead, they delimit the continuation of a handler that has already been selected. Their sole purpose is to record the exception that has been caught, such that |RETHROW| can access it inside such a block.
+
+.. note::
+   For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try block <exec-try-catch>` would be as follows.
+
+   Assume that :math:`\expand_F(bt) = [t1^n] \to [t2^m]`, for some :math:`n > m` such that :math:`t1^n[(n-m):n] = t2^m`,
+   and that the tag address `a` of :math:`x` corresponds to the tag type :math:`[t2^m] \to []`.
 
    .. math::
       \begin{array}{ll}
-      & \hspace{-5ex} S;~F;~\val^n~(\TRY~\X{bt}~\THROW~x~\CATCH~\RETURN~\END) \\
-      \stepto & S;~F;~\CATCHN_1\{\RETURN\}~(\LABEL_1 \{\}~\val^n~\THROWADDR~a~\END)~\END \\
+      & \hspace{-5ex} S;~F;~\val^n~(\TRY~\X{bt}~(\THROW~x)~\CATCH~x~\RETURN~\END) \\
+      \stepto & S;~F;~\LABEL_m\{\} (\CATCHadm\{a~\RETURN\}~\val^n~(\THROW~x)~\END)~\END \\
       \end{array}
 
-   :ref:`Handling <exec-throwaddr>` the thrown exception address :math:`a` in the throw context
-   :math:`T=\LABEL_1 \{\}[\_]~\END`, with the exception handler :math:`\CATCHN_1\{\RETURN\}` gives:
+   Denote :math:`\val^n = \val^{n-m} val^m`.
+   :ref:`Handling the thrown exception <exec-throwadm>` with tag address :math:`a` in the throw context
+   :math:`T=[val^{n-m}\_]`, with the exception handler :math:`H=\CATCHadm\{a~\RETURN\}` gives:
 
    .. math::
       \begin{array}{lll}
-      \stepto & S;~F;~\LABEL_1~\{\}~(\REFEXNADDR~a~\val^n)~\RETURN~\END & \hspace{9ex}\ \\
-      \stepto & (\REFEXNADDR~a~\val^n) & \\
+      \stepto & S;~F;~\LABEL_m\{\}~(\CAUGHTadm\{a~\val^m\}~\val^m~\RETURN~\END)~\END & \hspace{9ex}\ \\
+      \stepto & \val^m & \\
```

Comment:
I still don't really understand this example. I tried to understand this as a simple case, let's say, when `n == 3` and `m == 2`. Then after we throw two values (because the tag's signature is `t^m`) we still have one value (`n-m`, which means `3-2` here) value remaning. Where does that value go in this equation? This equation only has `m` and not `n`.

Can you guide me through this with a simple example, say, `n == 3`, `n == 2`, and `t^n == i32 f32 i64` and `t^m == f32 i64` or something?

Reply:
Thank you for this suggestion, the example will be nicer with such concrete values. I will update it in a followup PR.